src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,1053 @@
+    EVENT_LOOKFOROPENING_0              = 5,
+    EVENT_LOOKFOROPENING_1              = 6,
+    EVENT_LOOKFOROPENING_2              = 7,
+    EVENT_SUMMON_HAILSTONE              = 8,
+    EVENT_SUMMON_COLDWEAVE              = 9,
+    EVENT_SUMMON_FROSTWIND              = 10,
+    EVENT_SUMMON_AHUNE                  = 11,
+    EVENT_CLOSE_OPENING                 = 12,
+    EVENT_AHUNE_PHASE_ONE               = 13,
+    EVENT_AHUNE_PHASE_TWO               = 14,
+    EVENT_START_LOOKING_FOR_OPENING     = 15,
+    EVENT_STOP_LOOKING_FOR_OPENING      = 16
+};
+
+enum Actions
+{
+    ACTION_START_EVENT                  = -2574500,
+    ACTION_AHUNE_RETREAT                = -2586500,
+    ACTION_AHUNE_RESURFACE              = -2586501,
+    ACTION_EMOTE_RESURFACE              = -2575400
+};
+
+enum Phases
+{
+    PHASE_ONE                   = 0,
+    PHASE_TWO                   = 1
+};
+
+enum Points
+{
+    POINT_FLAMECALLER_000,
+    POINT_FLAMECALLER_001,
+    POINT_FLAMECALLER_002
+};
+
+enum Misc
+{
+    MAX_FLAMECALLERS    = 3
+};
+
+Position const SummonPositions[] =
+{
+    { -99.1021f, -233.7526f, -1.22307f, 1.588250f },  // Ahune
+    { -98.0151f, -230.4555f, -1.21089f, 1.797689f },  // Frozen Core
+    { -143.172f, -147.6801f, -3.16113f, 4.852015f },  // Bonfire Bunny 000
+    { -134.304f, -145.7803f, -1.70332f, 4.677482f },  // Bonfire Bunny 001
+    { -125.036f, -144.2065f, -1.91660f, 4.991642f },  // Bonfire Bunny 002
+    { -69.8121f, -162.4954f, -2.30451f, 1.710423f },  // Wisp Source Bunny
+    { -98.1029f, -230.7864f, -10.8085f, 1.448623f }   // Wisp Dest Bunny
+};
+
+Position const FlameCallerSpots[] =
+{
+    { -145.2233f, -137.5543f, -1.59056f, 5.427049f },
+    { -137.4383f, -136.4050f, -1.72384f, 5.336747f },
+    { -129.0413f, -132.1494f, -2.09285f, 5.460842f }
+};
+
+class boss_ahune : public CreatureScript
+{
+public:
+    boss_ahune() : CreatureScript("boss_ahune") { }
+
+    struct boss_ahuneAI : public BossAI
+    {
+        boss_ahuneAI(Creature* creature) : BossAI(creature, DATA_AHUNE)
+        {
+            Initialize();
+        }
+
+        void Initialize()
+        {
+            me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_KNOCK_BACK, true);
+            me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_KNOCK_BACK_DEST, true);
+            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE);
+        }
+
+        void EnterCombat(Unit* /*who*/) override
+        {
+            _EnterCombat();
+            events.ScheduleEvent(EVENT_INITIAL_EMERGE, 4);
+            events.ScheduleEvent(EVENT_SYNCH_HEALTH, 3000);
+        }
+
+        void EnterEvadeMode() override
+        {
+            if (Creature* ahuneBunny = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_AHUNE_BUNNY)))
+                ahuneBunny->AI()->EnterEvadeMode();
+            summons.DespawnAll();
+            me->DespawnOrUnsummon();
+        }
+
+        void JustDied(Unit* /*killer*/) override
```

Comment:
why there is no _JustDied() call inside that method? Otherwise boss state will stuck on IN_PROGRESS